### PR TITLE
Support custom name and description for Afterpay

### LINKED ISSUE #3391: The customized name and description of the Afterpay payment method are not reflected on the checkout page
**Describe the bug**
When changing the name and description of the Afterpay payment method, the changes are saved but not shown on the checkout page.

**To Reproduce**
1. Go to **WooCommerce > Settings > Payments > Stripe > Payment Methods**.
2. Click on the **Customize** button next to **Afterpay**.
3. Change the name and description. Click on the **Save changes** button.
4. Add a product to the cart and proceed to the checkout page.
5. The customized name and description *are not shown* for the Afterpay payment method.

**Expected behavior**
The customized name and description *are shown* for the Afterpay payment method.

**Screenshots**
Screenshot on 2024-08-29 at 12-33-49

Screenshot on 2024-08-29 at 12-34-06

**Environment (please complete the following information):**
 - WordPress Version: 6.6.1
 - WooCommerce Version: 9.2.3
 - Stripe Plugin Version: 8.6.1
 - Browser [e.g. chrome, safari] and Version: Firefox 129.0.2 (64-bit)
 - Any other plugins installed: -

**Additional context**
```
### WordPress Environment ###

WordPress address (URL): https://prinwp-test-2.mystagingwebsite.com
Site address (URL): https://prinwp-test-2.mystagingwebsite.com
WC Version: 9.2.3
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.8.1
Log Directory Writable: ✔
WP Version: 6.6.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.2.22
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 8.7.1
OpenSSL/3.0.13

SUHOSIN Installed: –
MySQL Version: 10.6.17-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 9.2.3
WC Database Prefix: wp_
Total Database Size: 5.69MB
Database Data Size: 3.02MB
Database Index Size: 2.67MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.08MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_abandoned_carts: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_automatewoo_customers: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_automatewoo_customer_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_guests: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_automatewoo_guest_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_log_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_queue: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_queue_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.13MB + Index: 0.09MB + Engine InnoDB
wp_jetpack_sync_queue: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 1.13MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.09MB + Index: 0.06MB + Engine InnoDB
wp_say_what_strings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_snippets: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availability: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availabilitymeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_booking_relationships: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.05MB + Index: 0.11MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.16MB + Index: 0.36MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 8
post: 1
product: 2
shop_order_placehold: 62
wp_font_face: 33
wp_font_family: 12
wp_global_styles: 1
wp_navigation: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (2) ###

WooCommerce Stripe Gateway: by WooCommerce – 8.6.1
WooCommerce: by Automattic – 9.2.3

### Inactive Plugins (13) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.3
AutomateWoo: by WooCommerce – 6.0.27 (update to version 6.0.33 is available)
Code Snippets: by Code Snippets Pro – 3.6.5.1
Jetpack: by Automattic – 13.8-a.7
Jetpack Protect: by Automattic - Jetpack Security team – 3.0.0
Pressable OnePress Login: by Pressable – 1.3.2
Say What?: by Ademti Software – 2.2.3
WooCommerce.com Update Manager: by Automattic – 1.0.3
WooCommerce Bookings: by WooCommerce – 2.1.5 (update to version 2.1.6 is available)
WooCommerce Gateway Affirm: by WooCommerce – 2.4.1
WooCommerce Shipping & Tax: by WooCommerce – 2.7.0
WooPayments: by WooCommerce – 8.0.1 (update to version 8.0.2 is available)
Woo Subscriptions: by WooCommerce – 6.2.0 (update to version 6.6.1 is available)

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 207 KB

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/ -  Contains the woocommerce/cart block
Checkout: #7 - /checkout/ -  Contains the woocommerce/checkout block
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Four
Version: 1.1 (update to version 1.2 is available)
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /wordpress/plugins/woocommerce/9.2.3/templates/block-notices/error.php
/wordpress/plugins/woocommerce/9.2.3/templates/block-notices/notice.php
/wordpress/plugins/woocommerce/9.2.3/templates/block-notices/success.php


### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: experimental-blocks
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system

Daily Cron: ✔ Next scheduled: 2024-08-29 08:51:36 +00:00
Options: ✔
Notes: 79
Onboarding: -

### Action Scheduler ###

Canceled: 1
Oldest: 2024-08-03 11:41:57 +0000
Newest: 2024-08-03 11:41:57 +0000

Complete: 83
Oldest: 2024-07-29 07:16:49 +0000
Newest: 2024-08-29 05:24:13 +0000

Failed: 21
Oldest: 2024-04-19 09:15:36 +0000
Newest: 2024-08-29 05:19:50 +0000

Pending: 3
Oldest: 2024-08-30 05:19:50 +0000
Newest: 2024-09-02 11:45:37 +0000


### Status report information ###

Generated at: 2024-08-29 05:35:54 +00:00
```

### PR DESCRIPTION
**Pull request by annemirasol**
Fixes #3391 

## Changes proposed in this Pull Request:
If the merchant provides a custom name and/or description for Afterpay, we want this to be shown in the checkout page instead of the default name and description. This is consistent with how other Stripe payment methods behave.

- Tweak the `get_title()` method override to inherit parent method behavior that takes custom  name into account.
- Delete the `get_description()` method override, as the default description is no longer needed, per https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3282.
- Added unit test

## Testing instructions
1. Enable Afterpay: `wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=methods`
2. As a shopper, add a product to cart and go to checkout.
3. Verify that the Afterpaymethod displays the default name (i.e. "Clearpay" in GB, "Afterpay" everywhere else)
4. Verify that there is no description displayed.
5. Go back to the Stripe payment methods page, click "Customize" and add a custom name and description for Afterpay: `wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=methods`
6. Go back to the checkout page and verify you can see the custom name. Select Afterpay to expand it and see the custom description.

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [x] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

Screenshot 1: Afterpay supporting custom name and description
Screenshot 2024-08-30 at 12 21 27 PM

### CODE REVIEW

File: includes/payment-methods/class-wc-stripe-upe-payment-method-afterpay-clearpay.php [hunk 1]
```diff
@@ -70,27 +70,11 @@ public function __construct() {
...
-		if ( $this->is_gb_country() ) {
-			return __(
-				'Allow customers to pay over time with Clearpay.',
-				'woocommerce-gateway-stripe'
-			);
-		}
-		return __(
-			'Allow customers to pay over time with Afterpay.',
-			'woocommerce-gateway-stripe'
-		);
```
Comment (annemirasol) [inline includes/payment-methods/class-wc-stripe-upe-payment-method-afterpay-clearpay.php:93]: Per https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3282, we no longer display the default description if a custom description is not provided. We delete this method override as it is no longer necessary.

File: tests/phpunit/test-class-wc-stripe-upe-payment-method.php [hunk 2]
```diff
@@ -615,6 +615,59 @@ public function test_payment_methods_are_reusable_if_cart_contains_subscription(
...
+		$p24_method               = $this->mock_payment_methods['p24'];
+		$eps_method               = $this->mock_payment_methods['eps'];
+		$sepa_method              = $this->mock_payment_methods['sepa_debit'];
+		$sofort_method            = $this->mock_payment_methods['sofort'];
+		$bancontact_method        = $this->mock_payment_methods['bancontact'];
+		$ideal_method             = $this->mock_payment_methods['ideal'];
+		$boleto_method            = $this->mock_payment_methods['boleto'];
+		$multibanco_method        = $this->mock_payment_methods['multibanco'];
+		$oxxo_method              = $this->mock_payment_methods['oxxo'];
+		$wechat_pay_method        = $this->mock_payment_methods['wechat_pay'];
```
Comment (Mayisha) [inline tests/phpunit/test-class-wc-stripe-upe-payment-method.php]: It looks like these are not used later in the test. Shall we remove these?

**Review by Mayisha — APPROVED** [review]
Thanks for the fix @annemirasol. I can see the correct title and description for AfterPay/ClearPay on the checkout page 👍 

- In `develop` my custom title/description is not rendered

Screenshot 2024-09-01 at 1 02 07 PM


- In `fix/3391-afterpay-custom-name-and-desc` branch my custom title/description is rendered correctly

**AfterPay**

Screenshot 2024-09-01 at 1 04 18 PM

**ClearPay**
Screenshot 2024-09-01 at 8 07 38 PM

File: includes/payment-methods/class-wc-stripe-upe-payment-method-afterpay-clearpay.php [hunk 3]
```diff
@@ -70,27 +70,11 @@ public function __construct() {
 	 */
 	public function get_title( $payment_details = false ) {
 		if ( $this->is_gb_country() ) {
-			return __( 'Clearpay', 'woocommerce-gateway-stripe' );
+			$this->title = __( 'Clearpay', 'woocommerce-gateway-stripe' );
```
Comment (wjrosa) [inline includes/payment-methods/class-wc-stripe-upe-payment-method-afterpay-clearpay.php:73]: `$this->title` is overwritten immediately below. You need to either include an `else` clause or return early.

**Review by wjrosa — APPROVED** [review]
Change looks good! Left a minor issue with the default payment method title. I guess it is a blocker, but approving it already.

File: tests/phpunit/test-class-wc-stripe-upe-payment-method.php (hunk 2 above)
Comment (annemirasol) [reply tests/phpunit/test-class-wc-stripe-upe-payment-method.php]: Yes, good catch, thank you! These can be cleaned up 🧹.

File: includes/payment-methods/class-wc-stripe-upe-payment-method-afterpay-clearpay.php (hunk 3 above)
Comment (annemirasol) [reply includes/payment-methods/class-wc-stripe-upe-payment-method-afterpay-clearpay.php:73]: Oof, good catch, thank you! Fixing... 🔧

**Pull request merged**